### PR TITLE
Lib creation

### PR DESCRIPTION
**Pull request by alexander-gorshenev**

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/library/FileUtil.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/library/FileUtil.kt]: Давай лучше https://github.com/JetBrains/kotlin-native/blob/master/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/util/NoJavaUtil.kt использовать.

File: dependencies/build.gradle [hunk 2]
```diff
@@ -178,16 +178,17 @@ class HelperNativeDep extends TgzNativeDep {
...
-task hostLibffi(type: HelperNativeDep) {
-    baseName = "libffi-3.2.1-2-$host"
-}
-
-task llvm(type: HelperNativeDep) {
-    baseName = "clang-llvm-$llvmVersion-$host"
-}
-
 if (isLinux()) {
+    // TODO this is the same as below.
```
Comment (olonho) [inline dependencies/build.gradle]: А почему так странно вышло?

**Comment by AdrianVovk** [comment]
Why is the commit so old? Why aren't you merging in the latest branch?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/library/FileUtil.kt (hunk 1 above)
Comment (alexander-gorshenev) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/library/FileUtil.kt]: Добавил коммит.

**Comment by alexander-gorshenev** [comment]
@AdrianVovk git is exceptionally good at tracking the dates, even when it doesn't make much sense. So the date of the very first commit I've done when first approached the feature has, as a steadfast tin soldier, survived multiple rebases and fixup squashes.

**Comment by AdrianVovk** [comment]
@alexander-gorshenev Oh that makes sense. Great to know.

I'd like to test this. How would I go about doing that?

**Comment by wenerme** [comment]
Any design doc about the lib ?

**Pull request merged**